subprojects/kotlin-dsl-tooling-builders/src/main/kotlin/org/gradle/kotlin/dsl/tooling/builders/KotlinBuildScriptModelBuilder.kt

```diff
@@ -486,7 +486,7 @@ inline fun KotlinScriptClassPathProvider.safeCompilationClassPathOf(
 } catch (error: Exception) {
     getSettings().run {
         serviceOf<ClassPathModeExceptionCollector>().collect(error)
-        compilationClassPathOf(rootClassLoaderScope)
+        compilationClassPathOf(baseProjectClassLoaderScope())
```

Comment:
:x: this should remain `rootClassLoaderScope` for "Settings scripts" and "Settings script plugins" in order to not include `buildSrc` ; and be changed to `settings.baseProjectClassLoaderScope()` for "Project scripts" and "Project script plugins"

Reply:
Done.